Submitted By: Jim Gifford (jim at linuxfromscratch dot org)
Date: 2005-04-24
Initial Package Version: 2.6
Origin: Gentoo and Self
Upstream Status: N/A
Description: Fixes Compile Issues with the 2.6 Kernel
 
diff -Naur net-tools-1.60.orig/hostname.c net-tools-1.60/hostname.c
--- net-tools-1.60.orig/hostname.c	2001-04-08 17:04:23.000000000 +0000
+++ net-tools-1.60/hostname.c	2004-06-24 06:17:32.517305695 +0000
@@ -42,10 +42,16 @@
 #include "config.h"
 #include "version.h"
 #include "../intl.h"
+#include <linux/version.h>
 
 #if HAVE_AFDECnet
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 5, 0)
 #include <netdnet/dn.h>
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 0)
+#include <linux/dn.h>
+#endif
+#endif
 
 char *Release = RELEASE, *Version = "hostname 1.100 (2001-04-14)";
 
diff -Naur net-tools-1.60.orig/lib/x25_sr.c net-tools-1.60/lib/x25_sr.c
--- net-tools-1.60.orig/lib/x25_sr.c	2000-05-20 13:38:10.000000000 +0000
+++ net-tools-1.60/lib/x25_sr.c	2004-06-24 06:15:45.163773724 +0000
@@ -22,6 +22,7 @@
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <linux/x25.h>
+#include <linux/version.h>
 #include <ctype.h>
 #include <errno.h>
 #include <netdb.h>
@@ -77,7 +78,11 @@
   rt.sigdigits=sigdigits;
 
   /* x25_route_struct.address isn't type struct sockaddr_x25, Why? */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0)
   memcpy(&rt.address, &sx25.sx25_addr, sizeof(x25_address));
+#else
+  memcpy(&rt.address, &sx25.sx25_addr, sizeof(struct x25_address));
+#endif
 
   while (*args) {
 	if (!strcmp(*args,"device") || !strcmp(*args,"dev")) {
diff -Naur net-tools-1.60.orig/iptunnel.c net-tools-1.60/iptunnel.c
--- net-tools-1.60.orig/iptunnel.c	2001-04-08 10:04:23.000000000 -0700
+++ net-tools-1.60/iptunnel.c	2006-04-25 01:29:12.416147208 -0700
@@ -25,16 +25,9 @@
 #include <fcntl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <netinet/in.h>
-#include <netinet/ip.h>
 #include <arpa/inet.h>
-#if defined(__GLIBC__) && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 1))
-#include <net/if.h>
-#include <net/if_arp.h>
-#else
 #include <linux/if.h>
 #include <linux/if_arp.h>
-#endif
 #include <linux/types.h>
 #include <linux/if_tunnel.h>
 
